### PR TITLE
Experiment/mapl 250 v2

### PR DESCRIPTION
**Pull request by sdeastham**
This update is part of the merge of MAPL 2.5.0.

### CODE REVIEW

**Comment by lizziel** [comment]
Hmm, I did not need to update ESMA_cmake for the MAPL update from v2.2.7 to v2.6.3 so it is still at v1.0.6 in main. While it should be updated to a more recent version soon (latest is v3.3.7) it does not seem necessary for MAPL. Did you find you needed this?

**Comment by sdeastham** [comment]
Not sure - I did things rather en-bloc, as I had some issues with MAPL which I hoped would be resolved by ESMA_cmake (but which may not have been). I'd certainly recommend that we do the update sooner rather than later as things seem to be diverging - most significantly the move towards shared libraries - but if you're not seeing any issues with using the older version then it seems lower priority for sure.

**Comment by LiamBindle** [comment]
I think it's okay if we hold off on updating ESMA_cmake right now. I think it would make sense to update at the same time as when I revert the original ESMA_cmake shim and replace it with and new version which uses GMAO's new ESMA_cmake that doens't have hard dependencies on Baselibs.

File: GNU.cmake [hunk 1]
```diff
@@ -120,30 +120,41 @@ set (common_Fortran_fpe_flags "-ffpe-trap=zero,overflow ${TRACEBACK} ${MISMATCH}
 
 # GEOS Debug
 # ----------
-set (GEOS_Fortran_Debug_Flags "${FOPT0} ${DEBINFO} -fcheck=all,no-array-temps -finit-real=snan")
+set (GEOS_Fortran_Debug_Flags "${FOPT0} ${DEBINFO} -fcheck=all,no-recursion,no-array-temps -finit-real=snan")
```
Comment (LiamBindle) [inline GNU.cmake:123]: New compiler option here

File: GNU.cmake [hunk 2]
```diff
@@ -120,30 +120,41 @@ set (common_Fortran_fpe_flags "-ffpe-trap=zero,overflow ${TRACEBACK} ${MISMATCH}
...
+# GEOS Aggressive
+# ---------------
+# NOTE: gfortran does get a benefit from vectorization, but the resulting code
+#       does not layout regress.
 
 # Options per Jerry DeLisle on GCC Fortran List
-#set (GEOS_Fortran_Vect_Flags "${FOPT2} -march=native -ffast-math -ftree-vectorize -funroll-loops --param max-unroll-times=4 -mprefer-avx128 -mno-fma")
-#set (GEOS_Fortran_Vect_FPE_Flags "${DEBINFO} ${TRACEBACK}")
+set (GEOS_Fortran_Aggressive_Flags "${FOPT2} -march=native -ffast-math -ftree-vectorize -funroll-loops --param max-unroll-times=4 -mprefer-avx128 -mno-fma")
+set (GEOS_Fortran_Aggressive_FPE_Flags "${DEBINFO} ${TRACEBACK} ${MISMATCH} ${ALLOW_BOZ}")
```
Comment (LiamBindle) [inline GNU.cmake:152]: Some new options here (in this file)

File: Intel.cmake [hunk 3]
```diff
@@ -46,6 +48,7 @@ set (ARCH_CONSISTENCY "-fimf-arch-consistency=true")
 set (FTZ "-ftz")
 set (ALIGN_ALL "\"SHELL:-align all\"")
 set (NO_ALIAS "-fno-alias")
+set (USE_SVML "-fimf-use-svml=true")
```
Comment (LiamBindle) [inline Intel.cmake:51]: There are some new Intel compiler flags (in this file)

File: Intel.cmake [hunk 4]
```diff
@@ -75,5 +80,11 @@ set (GEOS_Fortran_Release_FPE_Flags "${common_Fortran_fpe_flags} ${ARCH_CONSISTE
 set (GEOS_Fortran_Vect_Flags "${FOPT3} ${DEBINFO} -xCORE-AVX2 -fma -qopt-report0 ${FTZ} ${ALIGN_ALL} ${NO_ALIAS} -align array32byte")
 set (GEOS_Fortran_Vect_FPE_Flags "${FPE3} ${FP_MODEL_CONSISTENT} ${NOOLD_MAXMINLOC}")
 
+# GEOS Aggressive
+# ---------------
+set (GEOS_Fortran_Aggressive_Flags "${FOPT3} ${DEBINFO} -xCORE-AVX2 -fma -qopt-report0 ${FTZ} ${ALIGN_ALL} ${NO_ALIAS} -align array32byte")
```
Comment (LiamBindle) [inline Intel.cmake:85]: Some new compiler flags here too (in this file)

**Comment by LiamBindle** [comment]
I think we should keep in mind that GMAO has updated some compiler settings for GEOS (which we copy for MAPL and GEOS-Chem). If we run into runtime errors/compiler oddities, it might be worth revisiting these new compiler flags (i.e., we might be missing a flag or two that they are using).  I think this is the most relevant detail to keep in mind if we hold off on upgrading for a little bit.

I've marked updates to the compiler flags above, for furture reference.

**Comment by lizziel** [comment]
@LiamBindle and @sdeastham, can this PR be closed?

**Comment by LiamBindle** [comment]
Yes, this shouldn't be necessary any more. We can use the official GMAO ESMA_cmake now.

**Pull request closed without merging**